### PR TITLE
Update g2c recipe: add develop version and misc. variants

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
This PR adds a develop version for g2c, adds miscellaneous variants, and adds logic for running ctest through `spack install --test`.

### CODE REVIEW

File: var/spack/repos/builtin/packages/g2c/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/g2c/package.py:27]: Thanks for moving these directives to AFTER the versions.

**Review by tldahlgren — APPROVED** [review]
Confirmed the branch exists.

**Comment by tldahlgren** [comment]
> Confirmed the branch exists.

Successfully installed the software with `--test=root`, obtaining the following test results:

```
$ cat opt/spack/linux-rhel8-x86_64_v3/gcc-12.1.1/g2c-1.7.0-wk7v546gwerxub4ns2nkwhflfflvsgfb/.spack/install-time-test-log.txt
==> Testing package g2c-1.7.0-wk7v546
==> [2023-09-12-14:02:02.870199] Running build-time tests
==> [2023-09-12-14:02:02.870409] RUN-TESTS: build-time tests [check]
==> [2023-09-12-14:02:02.871481] '/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel8-x86_64_v3/gcc-12.1.1/gmake-4.4.1-56tm3r7fo3dypyw2nzzdrvpssqxaorfj/bin/make' '-j16' 'test'
Running tests...
/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel8-x86_64_v3/gcc-12.1.1/cmake-3.26.3-xhxkjrk6lpvyajpdocyw7q5y2yfrfnre/bin/ctest --force-new-ctest-process 
Test project /tmp/dahlgren/spack-stage/spack-stage-g2c-1.7.0-wk7v546gwerxub4ns2nkwhflfflvsgfb/spack-build-wk7v546
      Start  1: tst_g2_create
 1/21 Test  #1: tst_g2_create ....................   Passed    0.01 sec
      Start  2: tst_g2_addgrid
 2/21 Test  #2: tst_g2_addgrid ...................   Passed    0.00 sec
      Start  3: tst_g2_addfield
 3/21 Test  #3: tst_g2_addfield ..................   Passed    0.00 sec
      Start  4: tst_decode
 4/21 Test  #4: tst_decode .......................   Passed    0.00 sec
      Start  5: tst_gbits
 5/21 Test  #5: tst_gbits ........................   Passed    0.01 sec
      Start  6: tst_gridtemplates
 6/21 Test  #6: tst_gridtemplates ................   Passed    0.00 sec
      Start  7: tst_pdstemplates
 7/21 Test  #7: tst_pdstemplates .................   Passed    0.00 sec
      Start  8: tst_drstemplates
 8/21 Test  #8: tst_drstemplates .................   Passed    0.01 sec
      Start  9: tst_unpack
 9/21 Test  #9: tst_unpack .......................   Passed    0.00 sec
      Start 10: tst_getdim
10/21 Test #10: tst_getdim .......................   Passed    0.00 sec
      Start 11: tst_simpack
11/21 Test #11: tst_simpack ......................   Passed    0.00 sec
      Start 12: tst_addfield_spec
12/21 Test #12: tst_addfield_spec ................   Passed    0.01 sec
      Start 13: tst_spec
13/21 Test #13: tst_spec .........................   Passed    0.00 sec
      Start 14: tst_com
14/21 Test #14: tst_com ..........................   Passed    0.00 sec
      Start 15: tst_g2_addlocal
15/21 Test #15: tst_g2_addlocal ..................   Passed    0.01 sec
      Start 16: tst_seekgb
16/21 Test #16: tst_seekgb .......................   Passed    0.01 sec
      Start 17: tst_png
17/21 Test #17: tst_png ..........................   Passed    0.01 sec
      Start 18: tst_addfield2
18/21 Test #18: tst_addfield2 ....................   Passed    0.00 sec
      Start 19: tst_jpeg
19/21 Test #19: tst_jpeg .........................   Passed    0.01 sec
      Start 20: tst_addfield3
20/21 Test #20: tst_addfield3 ....................   Passed    0.01 sec
      Start 21: tst_addfield4
21/21 Test #21: tst_addfield4 ....................   Passed    0.01 sec

100% tests passed, 0 tests failed out of 21

Total Test time (real) =   0.17 sec
```

**Comment by tldahlgren** [comment]
Let me know whether I should wait for one of the other maintainers to comment before I merge this PR.

**Comment by AlexanderRichert-NOAA** [comment]
Let's go ahead and merge please, thanks!

**Pull request merged**
